server/base/src/main/java/org/apache/accumulo/server/conf/cluster/ClusterConfigParser.java

```diff
@@ -159,8 +159,12 @@ private static void validateConfiguredGroups(final ServerContext ctx, final Set<
       if (!zkGroups.contains(cg)) {
         if (createMissingRG) {
           try {
-            ctx.resourceGroupOperations().create(ResourceGroupId.of(cg));
-          } catch (AccumuloException | AccumuloSecurityException e) {
+            // cant use API as servers may not be up when this is called
+            // from accumulo-cluster
+            final ResourceGroupId rgid = ResourceGroupId.of(cg);
+            final ResourceGroupPropKey key = ResourceGroupPropKey.of(rgid);
+            key.createZNode(ctx.getZooSession().asReaderWriter());
+          } catch (KeeperException | InterruptedException e) {
```

Comment:
This isn't the best place for this code. It's out of scope of configuration file parsing, and will make it harder to replace this with `yq` in future.

There's also another place in the main method, shortly after it creates the ServerContext, where it uses the `context.resourceGroupOperations().list()` public API.

I think these features to modify ZK in this file should be stripped out, and instead, it should be added to the existing ZooPropEditor or related admin tool for managing property configuration nodes in ZK, where it would fit more naturally alongside the same options for table, namespace, and system configs. This class should stick to just parsing the configuration file.

Reply:
I can make this change, but I think it's going to effect the user experience. The user will configure `cluster.yaml` and run `accumulo-cluster start`, then when `ClusterConfigParser.validateConfiguredGroups` is called it will fail because the resource group have not been created. I can no longer create them in `accumulo-cluster` because I can't parse `cluster.yaml`. We'll have to kick out an error telling the user to go run `ZooPropEditor` to create the resource groups. Removing `ClusterConfigParser.validateConfiguredGroups` is likely not an option because if the `cluster.yaml` is misconfigured, then a bunch of resources won't start.

Reply:
> I guess, though I'm still confused about why a RG that has no config in ZK isn't valid.

The configuration node in ZK is currently how a resource group is known to exist. The ResourceGroup configuration is read from the node in ZooKeeper in ServerConfigurationFactory.

Reply:
PR updated based on this suggestion, not fully tested yet.